applications/vanilla/views/modules/promoted.php

```diff
@@ -0,0 +1,112 @@
+<h1 class="H"><?php echo $this->Data('Title'); ?></h1>
+<?php
+
+if ($data = $this->Data('Content')) {
+   if ($view = $this->Data('View') == 'table') {
+      writeTable($data);
+   } else {
+      writeList($data);
+   }
+} else {
+   echo $this->Data('EmptyMessage');
+}
+
+function writeList($data) {
+?>
+    <ul class="PromotedContentList DataList">
+       <?php foreach ($data as $row) {
+          writeRow($row, 'modern');
+       } ?>
+    </ul>
+<?php
+}
+
+function writeTable($data) { ?>
+   <div class="DataTableContainer">
+      <div class="DataTableWrap">
+         <table class="DataTable">
+            <thead>
+            <tr>
+               <td class="DiscussionName">
+                  <div class="Wrap"><?php echo T('Subject'); ?></div>
+               </td>
+               <td class="BlockColumn BlockColumn-User LastUser">
+                  <div class="Wrap"><?php echo T('Author'); ?></div>
+               </td>
+            </tr>
+            </thead>
+            <?php foreach ($data as $row) {
+               writeRow($row, 'table');
+            } ?>
+         </table>
+      </div>
+   </div>
+   <?php
+}
+
+function writeRow($row, $view) {
+   $title        =  htmlspecialchars(val('Name', $row));
+   $url          = val('Url', $row);
+   $body         = Gdn_Format::PlainText(val('Body', $row), val('Format', $row));
+   $categoryUrl  = val('CategoryUrl', $row);
+   $categoryName = val('CategoryName', $row);
+   $date         = val('DateUpdated', $row) ?: val('DateInserted', $row);
+   $date         = Gdn_Format::Date($date, 'html');
+   $type         = val('RecordType', $row, 'post');
+   $id           = val('CommentID', $row, val('DiscussionID', $row, ''));
+   $author       = val('Author', $row);
+   $username     = val('Name', $author);
+   $userUrl      = val('Url', $author);
+   $userPhoto    = val('PhotoUrl', $author);
+   $cssClass     = val('_CssClass', $author);
```

Comment:
Typically we use these `val`s in context rather than calling a big stack of them like this.